docs/references/ingress-messages.mdx

```diff
@@ -0,0 +1,82 @@
+---
+keywords: [intermediate, reference, security, ingress message, query call, update call, error handling, idempotency]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Ingress messages on ICP
+
+<MarkdownChipRow labels={["Intermediate","Reference", "Security"]} />
+
+## Overview
+
+Ingress messages are calls to ICP canisters that are sent by external entities, such as users, usually through an agent like [agent-js](https://github.com/dfinity/agent-js/tree/main) that runs within a web application, or [agent-rs](https://github.com/dfinity/agent-rs/tree/main) in a Rust application. See also the introductory [call overview](/docs/current/developer-docs/smart-contracts/call/overview). 
+
+For example, you can browse what ingress messages can be sent to the ckBTC ledger by browsing through its [API on the dashboard](https://dashboard.internetcomputer.org/canister/mxzaz-hqaaa-aaaar-qaada-cai). The tool also allows to send messages, using agent-js behind the scenes.
+
+In the following, we discuss ICP's ingress message APIs. While these APIs are defined in detail in the [HTTPS interface specification](/docs/current/references/ic-interface-spec#http-interface), we provide a more high level and intuitive overview. In addition to discussing the happy path, we lay a special focus on error handling. That aspect is particularly important as it can be tricky to determine if an ingress message has actually been successfully executed. In the worst case, misinterpreting errors could lead to bugs such as double spending.
```

Comment:
```suggestion
This page discusses ICP's ingress message APIs. While these APIs are defined in detail within the [HTTPS interface specification](/docs/current/references/ic-interface-spec#http-interface), this page provides a more high-level and intuitive overview, with a special focus on error handling. That aspect is particularly important, as it can be tricky to determine if an ingress message has actually been successfully executed. Misinterpreting errors could lead to bugs such as double spending.
```

Reply:
Done, thanks!